### PR TITLE
Bugfix in `mesh.sampling.find_nearest_cells` due to incorrect norm usage

### PR DESCRIPTION
**Pull request by peterdsharpe**
# PhysicsNeMo Pull Request

## Description

Fixed a bug in `mesh.sampling.find_nearest_cells`, where a mixup between L2 and L-inf norms could cause slightly incorrect nearest-neighbor assignments in highly skewed meshes.

Also adds a new test to guard against regression.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [x] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: physicsnemo/mesh/sampling/sample_data.py [hunk 1]
```diff
@@ -547,66 +547,78 @@ def _find_nearest_cells_bvh(
...
     resolved = torch.zeros(n_queries, dtype=torch.bool, device=device)
 
     ### Estimate initial search tolerance from BVH root extent
     root_extent = bvh.node_aabb_max[0] - bvh.node_aabb_min[0]
-    # Typical cell diameter ~ total extent / n_cells^(1/d)
     tolerance = root_extent.max().item() / max(n_cells ** (1.0 / n_spatial_dims), 1.0)
 
     ### Expanding-radius search: double tolerance each round until all resolved
-    max_rounds = 20  # tolerance doubles each round → covers 2^20 ~ 1M× initial
+    max_rounds = 25  # tolerance doubles each round → covers 2^25 ~ 33M× initial
```
Comment (laserkelvin) [inline physicsnemo/mesh/sampling/sample_data.py]: Do you want to just make this a default kwarg? Just for the sake of some tunability in either direction

File: physicsnemo/mesh/sampling/sample_data.py [hunk 2]
```diff
@@ -547,66 +547,78 @@ def _find_nearest_cells_bvh(
...
+            cell_indices[global_query[update_mask]] = tgt[update_mask]
 
-            # Mark queries that received at least one candidate as resolved
-            has_candidate = candidates.counts > 0
-            resolved[remaining_idx[has_candidate]] = True
+        # Resolve when the L-inf search radius guarantees no closer cell was missed
+        newly_resolved = (
+            (best_dist_sq < float("inf")) & ~resolved & (tolerance**2 >= best_dist_sq)
+        )
+        resolved |= newly_resolved
```
Comment (laserkelvin) [inline physicsnemo/mesh/sampling/sample_data.py]: I'm not super used to `|=` as syntactic sugar - might be good to leave a comment here to say what the operation is

File: physicsnemo/mesh/sampling/sample_data.py [hunk 3]
```diff
@@ -547,66 +547,78 @@ def _find_nearest_cells_bvh(
...
-        remaining_idx = torch.where(remaining_mask)[0]
+        remaining_idx = torch.where(~resolved)[0]
         if len(remaining_idx) == 0:
             break
 
         candidates = bvh.find_candidate_cells(
             query_points[remaining_idx],
             aabb_tolerance=tolerance,
-            max_candidates_per_point=64,
+            max_candidates_per_point=None,
```
Comment (laserkelvin) [inline physicsnemo/mesh/sampling/sample_data.py]: Is this safe to do? I would imagine maybe instead of setting it as `None` as default, maybe similar to `max_rounds` just set it as a kwarg and have it default to a large number instead of not setting a limit at all to not OOM

File: physicsnemo/mesh/sampling/sample_data.py (hunk 1 above)
Comment (peterdsharpe) [reply physicsnemo/mesh/sampling/sample_data.py]: Yeah, good call - might as well expose this

File: physicsnemo/mesh/sampling/sample_data.py (hunk 2 above)
Comment (peterdsharpe) [reply physicsnemo/mesh/sampling/sample_data.py]: I think this one follows from other in-place ops:

- `a += b` is roughly `a = a + b` but in-place.
- `a |= b` is roughly `a = a | b` but in-place.

I can add a comment to the line end though!

File: physicsnemo/mesh/sampling/sample_data.py (hunk 3 above)
Comment (peterdsharpe) [reply physicsnemo/mesh/sampling/sample_data.py]: Yeah - that would be better practice! Good call, will do

File: physicsnemo/mesh/sampling/sample_data.py (hunk 1 above)
Comment (peterdsharpe) [reply physicsnemo/mesh/sampling/sample_data.py]: Fixed

File: physicsnemo/mesh/sampling/sample_data.py (hunk 2 above)
Comment (peterdsharpe) [reply physicsnemo/mesh/sampling/sample_data.py]: Fixed

File: physicsnemo/mesh/sampling/sample_data.py (hunk 3 above)
Comment (peterdsharpe) [reply physicsnemo/mesh/sampling/sample_data.py]: Fixed
